src/util/subscribeToResult.ts

```diff
@@ -68,15 +67,18 @@ export function subscribeToResult<T>(outerSubscriber: OuterSubscriber<any, any>,
         break;
       }
     } while (true);
-  } else if (typeof result[$$observable] === 'function') {
+  } else if (result && typeof result[$$observable] === 'function') {
     const obs = result[$$observable]();
     if (typeof obs.subscribe !== 'function') {
-      destination.error(new Error('invalid observable'));
+      destination.error(new TypeError('Provided object does not correctly implement Symbol.observable'));
     } else {
       return obs.subscribe(new InnerSubscriber(outerSubscriber, outerValue, outerIndex));
     }
   } else {
-    destination.error(new TypeError('unknown type returned'));
+    const value = isObject(result) ? 'an invalid object' : `'${result}'`;
+    const msg = `You provided ${value} where a stream was expected.`
+      + ' You can provide an Observable, Promise, Array, or Iterable.';
```

Comment:
Yeah, I just punted on `Symbol.observable` because it felt like one of those "if you know it exists, you know it's supported" things. But I'm cool with adding `of an object implementing Symbol.observable`